tests/integration/software-trigger-acquires-single-frames.cpp

```diff
@@ -0,0 +1,194 @@
+    CHECK(runtime);
+    CHECK(dm);
+
+    AcquireProperties props = {};
+    OK(acquire_get_configuration(runtime, &props));
+
+    DEVOK(device_manager_select(dm,
+                                DeviceKind_Camera,
+                                SIZED("simulated: empty") - 1,
+                                &props.video[0].camera.identifier));
+    DEVOK(device_manager_select(dm,
+                                DeviceKind_Storage,
+                                SIZED("trash") - 1,
+                                &props.video[0].storage.identifier));
+
+    OK(acquire_configure(runtime, &props));
+
+    AcquirePropertyMetadata metadata = { 0 };
+    OK(acquire_get_configuration_metadata(runtime, &metadata));
+
+    props.video[0].camera.settings.binning = 1;
+    props.video[0].camera.settings.pixel_type = SampleType_u12;
+    props.video[0].camera.settings.shape = { .x = 1024, .y = 1024 };
+    props.video[0].max_frame_count = 10;
+
+    // Enable Software Trigger
+    {
+        int i_line = select_software_trigger_line(&metadata);
+        CHECK(props.video[0].camera.settings.input_triggers.frame_start.kind ==
+              Signal_Input);
+        props.video[0].camera.settings.input_triggers.frame_start.edge =
+          TriggerEdge_Rising;
+        props.video[0].camera.settings.input_triggers.frame_start.line = i_line;
+        props.video[0].camera.settings.input_triggers.frame_start.enable = 1;
+    }
+
+    OK(acquire_configure(runtime, &props));
+}
+
+static const VideoFrame*
+next(const VideoFrame* frame)
+{
+    const uint8_t* f = (const uint8_t*)frame;
+    return (const VideoFrame*)(f + frame->bytes_of_frame);
+}
+
+static int
+frame_count(const VideoFrame* beg, const VideoFrame* end)
+{
+    int i = 0;
+    for (const VideoFrame* cur = beg; cur < end; cur = next(cur))
+        ++i;
+    return i;
+}
+
+static bool
+is_running(AcquireRuntime* runtime)
+{
+    return acquire_get_state(runtime) == DeviceState_Running;
+}
+
+int
+main()
+{
+    float test_timeout_ms = 5000.0;
+    AcquireRuntime* runtime = 0;
+    try {
+        runtime = acquire_init(reporter);
+        setup(runtime);
+        OK(acquire_start(runtime));
+
+        struct clock t0;
+        clock_init(&t0);
+        int n = 0;
+        while (is_running(runtime)) {
+            VideoFrame *beg, *end;
+
+            // Haven't triggered yet, so expect no data
+            OK(acquire_map_read(runtime, 0, &beg, &end));
+            EXPECT(end == beg, "Expected no available data.");
+
+            OK(acquire_execute_trigger(runtime, 0));
+
+            // wait for data
+            while (end == beg && is_running(runtime) &&
+                   clock_toc_ms(&t0) < test_timeout_ms) {
+                // delay so loop isn't busy polling
+                clock_sleep_ms(0, 10);
+                OK(acquire_map_read(runtime, 0, &beg, &end));
+            }
+
+            if (end == beg && !is_running(runtime))
+                break;
```

Comment:
@andy-sweet I think the test is not flaky now. 

I kept the sleep in the loop just so it's not burning CPU. I had to add the additional `is_running` check. Otherwise, there was a window where the acquisition would stop but it would make it to line 167 and hit the test timeout (and fail).